cmd/corald/handlers/action.go

```diff
@@ -0,0 +1,92 @@
+package handlers
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+
+	"github.com/ardanlabs/kit/cfg"
+	"github.com/ardanlabs/kit/web"
+	"github.com/coralproject/shelf/internal/sponge/item"
+	"github.com/coralproject/shelf/internal/talk"
+)
+
+// actionHandle maintains the set of handlers for the action api.
+type actionHandle struct{}
+
+// Action fronts the access to the action service functionality.
+var Action actionHandle
+
+var (
+	// ErrParameterNotFound is action not found error.
+	ErrParameterNotFound = errors.New("Parameter not found")
+)
+
+//==============================================================================
+
+// Create an action (flag, like, etc) on an Item.
+// 200 Success, 404 Not Found, 500 Internal
+func (actionHandle) Create(c *web.Context) error {
+
+	spongedURL := cfg.MustURL("SPONGED_URL").String()
+
+	// Get the user's item_id.
+	var user item.Item
+	if err := json.NewDecoder(c.Request.Body).Decode(&user); err != nil {
+		return err
+	}
+
+	// Action to be applied to the target with targetID.
+	action, ok := c.Params["action"]
```

Comment:
I don't think we need the ok test here for action and item_key, as these are explicitly in the path. If they are not found something is really bad.